src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -63,24 +63,24 @@ constexpr bool contains_indices_to_contract(
  *
  * @tparam LhsTensorIndices the TensorIndexs of the Tensor on the LHS of the
  * tensor expression, e.g. `ti_a`, `ti_b`, `ti_c`
- * @tparam T the type of the RHS TensorExpression
  * @param rhs_te the RHS TensorExpression to be evaluated
- * @return the LHS Tensor with index order specified by LhsTensorIndices
+ * @return the resultant LHS Tensor with index order specified by
+ * LhsTensorIndices
  */
-template <auto&... LhsTensorIndices, typename T,
-          Requires<std::is_base_of<Expression, T>::value> = nullptr>
-auto evaluate(const T& rhs_te) {
+template <auto&... LhsTensorIndices, typename TE,
+          Requires<std::is_base_of<Expression, TE>::value> = nullptr>
+auto evaluate(const TE& rhs_te) {
```

Comment:
Is it possible to have this `evaluate` construct the tensor and then call the other? That could remove some of the code duplication in the checks. In other pieces of code we usually have the "return by value" function call the "return by not_null" function. This helps keep code duplication down

Reply:
Yep! Don't know why I didn't just do that in the first place.

Just want to check, then do we also want [this](https://github.com/sxs-collaboration/spectre/blob/d766ddab22b38f9bf8c026cab58666f0805d5e89/src/DataStructures/Tensor/Tensor.hpp#L146) `Tensor` constructor removed? I imagine the answer is "yes," but I didn't want to assume in case it was useful to keep around for some other purpose

Reply:
i.e. if the first `evaluate` overload just calls the second one and the second one does the filling, the first one doesn't need this constructor anymore

Reply:
I'd say remove the constructor if it's never called. That way people won't accidentally do something weird with the constructor :)

Reply:
Done in the second fixup!